cram/cram_io.c

```diff
@@ -2449,23 +2428,25 @@ static int cram_populate_ref(cram_fd *fd, int id, ref_entry *r) {
         if (strncmp(tag->str+3, md5_buf2, 32) != 0) {
             hts_log_error("Mismatching md5sum for downloaded reference");
             hclose_abruptly(fp);
-            unlink(path_tmp);
+            unlink(path_tmp.s);
+            free(path_tmp.s);
             return -1;
         }
 
         if (hwrite(fp, r->seq, r->length) != r->length) {
             perror(path);
         }
         if (hclose(fp) < 0) {
-            unlink(path_tmp);
+            unlink(path_tmp.s);
         } else {
-            if (0 == chmod(path_tmp, 0444))
-                rename(path_tmp, path);
+            if (0 == chmod(path_tmp.s, 0444))
+                rename(path_tmp.s, path);
```

Comment:
Can you also check for `rename()` error here and unlink the temp file in that case? Probably all three conditions `hclose(fp) < 0 || chmod(path_tmp.s, 0444) != 0 || rename(path_tmp.s, path) != 0` could be grouped into a single check resulting in `unlink(path_tmp.s)`.

Reply:
It's a pre-existing buglet, but sure — since I touched the code I guess I get to fix it :smile: